#!/usr/bin/env python

PACKAGE = "rviz_obstacle_generator"  # Replace with your package name

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
gen.add("infinite_obstacle", bool_t, 0, "Infinite_obstacle", True)

gen.add("obs_id", int_t, 0, "obstacle ID", 0, 0, 9)
gen.add("Kill", bool_t, 0, "Kill obstacle", False)
gen.add("lane_id", int_t, 0, "Lane ID", 0, 0, 2)
gen.add("NewLane", bool_t, 0, "NewLane obstacle", False)
gen.add("speed_km_p_h", double_t, 0, "speed of the dynamic obstacle", 40.0, 0.1, 120.0)
gen.add("NewSpeed", bool_t, 0, "NewSpeed obstacle", False)

gen.add("duration_sec_lane_following", double_t, 0, "duration time of the lane_following obstacle", 3.0, 0.0, 100.0)



# gen.add("obstacle_shape", str_t, 0, "Shape of the obstacle", "rectangle")

# obstacle_type_enum = gen.enum([ 
#     gen.const("ChangeSpeed", int_t, 0, "ChangeSpeed"),
#     gen.const("ChangeLane", int_t, 1, "ChangeLane"),
#     gen.const("KILL", int_t, 2, "KILL")
# ], "Type of the obstacle")

# gen.add("dynamic_config_state", int_t, 0, "Choose the type of obstacle", 0, 0, 10, edit_method=obstacle_type_enum)
# gen.add("Activate", bool_t, 0, "Activate functions", False)
gen.add("duration_sec_dynamic", double_t, 0, "duration time of the dynamic obstacle", 3.0, 0.0, 20.0)
gen.add("angular_velocity_deg_p_sec", double_t, 0, "angular velocity of the dynamic obstacle", 0.0, -30.0, 30.0)

exit(gen.generate(PACKAGE, "Obstaclev2", "Obstaclev2"))
